Algo.h

```diff
@@ -129,32 +159,93 @@ class MMP
         // ps1.x()));
         double x = 0.0;
         // (x, 0) is equidistant from ps1 and ps2
+        //calculated x directly from research paper
+        double alpha = ps2.x() - ps1.x();
+        double beta = i2.ps_d - i1.ps_d;
+        double gamma = ps1.squaredLength() - ps2.squaredLength() - beta*beta;
+        //ax^2 + bx + c = 0
+        double a = alpha*alpha - beta*beta;
+        //γα + 2s1xβ2
+        double b = gamma*alpha + 2.0*ps2.x()*beta*beta;
+        //c = 1/4γ2 − s12β2
+        double c = 0.25*gamma*gamma - ps2.squaredLength()*beta*beta;
+
+        if(b*b - 4*a*c > 0)
```

Comment:
I see that will happen, but there msut be a better way to write it rather than setting `x = -1`? Maybe set a bool flag `no_equidistant_pt`, and use it accordingly.